### PR TITLE
netpbm: 10.89.1 → 10.91.2

### PR DESCRIPTION
**Pull request by jtojnar**
###### Motivation for this change
Tried updating if it makes cross compilation better.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jtojnar** [comment]
While reading the build system sources, I noticed it also uses vendored Jasper. We might want to use the [Fedora patch](https://src.fedoraproject.org/rpms/netpbm/blob/master/f/netpbm-jasper.patch) to unbundle it or better find a way to disable it since Jasper is full of security holes.

**Comment by kampka** [comment]
Cross-building works for the most part, but I found some iregularitues with shebangs, eg.

```
$ nix-build . -A pkgsCross.aarch64-multiplatform.netpbm
$ head result-bin/bin/ppmquant                                                               
#! /nix/store/j206pskg2yzyla1cnfrb9kb5n5bfvjgm-bash-4.4-p23/bin/bash -e
```

Otherwise, nicely done :+1:

**Comment by jtojnar** [comment]
Could you clarify what the irregularities are?

**Comment by kampka** [comment]
> Could you clarify what the irregularities are?

Of course. After cross building from x86 ot aarch64, the shebang should point to an aarch64 version of bash, otherwise the script would probably run into an exec error.
The bash it points to however is the one of the build system
```
$ file /nix/store/j206pskg2yzyla1cnfrb9kb5n5bfvjgm-bash-4.4-p23/bin/bash
/nix/store/j206pskg2yzyla1cnfrb9kb5n5bfvjgm-bash-4.4-p23/bin/bash: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /nix/store/2pi6zgkwnr3zdslvlv16nixpzvbyjx1n-glibc-2.31/lib/ld-linux-x86-64.so.2, for GNU/Linux 2.6.32, not stripped
```

**Comment by jtojnar** [comment]
Weird. I would expect the patchShebangsAuto hook to do the right thing 
wrt cross compilation. OR is the wrapProgram causing the issue?

**Comment by jtojnar** [comment]
I tried using `buildPackages.makeWrapper` hoping it would fix the 
interpolated shell variable

https://github.com/NixOS/nixpkgs/blob/b0a8a746b3d0edebf320b9923ba1cc8fefa0ee00/pkgs/top-level/all-packages.nix#L456-L457

but it produces the same derivation.

**Comment by kampka** [comment]
> Weird. I would expect the patchShebangsAuto hook to do the right thing wrt cross compilation. OR is the wrapProgram causing the issue?

It looks like `patchShebang` does [the wrong thing](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/setup-hooks/patch-shebangs.sh#L89) here as the bash reference is already a reference to the Nix store.
This would probably be worth fixing at that level.

**Comment by jtojnar** [comment]
That is expected. The operation should be idempotent.

We need to convine wrapProgram to use the correct shebang from the 
start.

**Comment by kampka** [comment]
A practical solution (not necessarily pretty) would be to just patch it over [like so](https://github.com/NixOS/nixpkgs/pull/96959/files#diff-ce8581c807ea0e55a715459e92049bafR25).

File: pkgs/tools/graphics/netpbm/default.nix [hunk 1]
```diff
@@ -40,14 +42,25 @@ stdenv.mkDerivation {
...
 
+  # Environment variables
+  STRIPPROG = "${stdenv.lib.getBin stdenv.cc.bintools.bintools}/bin/${stdenv.cc.targetPrefix}strip";
+
   postPatch = ''
     # Install libnetpbm.so symlink to correct destination
     substituteInPlace lib/Makefile \
       --replace '/sharedlink' '/lib'
+
+    # TODO: move to config.mk in 10.91.4
```
Comment (jtojnar) [inline pkgs/tools/graphics/netpbm/default.nix:61]: Does not seem to be part of 10.91.4, even though the [commit](https://sourceforge.net/p/netpbm/code/3941/) is older than the [release](https://sourceforge.net/p/netpbm/code/3962/).

Comment (jtojnar) [reply pkgs/tools/graphics/netpbm/default.nix:61]: Confirmed with upstream that it is expected to be in 10.92 (~two weeks).

**Comment by jtojnar** [comment]
@GrahamcOfBorg build netpbm `pkgsCross.aarch64-multiplatform.netpbm`

**Comment by jtojnar** [comment]
@Ericson2314 @matthewbauer any idea how to make `makeWrapper` use the correct shell for shebang?

**Comment by Ericson2314** [comment]
@jtojnar yeah I recently learned that `makeWrapper is set up wrong. it should use `targetPackages.runtimeShell` so we use `depsBuildBuild` vs `nativeBuildInputs` to control what shell is used. Instead people have been putting it in `buildInputs` to achieve the same result which isn't right when its only a build-time dep.

I should make an issue for this.

**Pull request merged**